include/rcpputils/filesystem_helper.hpp

```diff
@@ -352,6 +360,58 @@ RCPPUTILS_PUBLIC bool operator!=(const path & a, const path & b);
 */
 RCPPUTILS_PUBLIC std::ostream & operator<<(std::ostream & os, const path & p);
 
+// remove warning suppression
+#if !defined(_WIN32)
+# pragma GCC diagnostic pop
+#else  // !defined(_WIN32)
+# pragma warning(pop)
+#endif
+
+/**
+ * \brief Get a path to a location in the temporary directory, if it's available.
+ *
+ * This does not create any directories.
+ * On Windows, this uses "GetTempPathA"
+ * On non-Windows, this prefers the environment variable TMPDIR, falling back to /tmp
+ *
+ * \return A path to a directory for storing temporary files and directories.
+ */
+RCPPUTILS_PUBLIC std::filesystem::path temporal_directory_path();
+
+/**
+ * \brief Construct a uniquely named temporary directory, in "parent", with format base_nameXXXXXX
+ *
+ * The output, if successful, is guaranteed to be a newly-created directory.
+ * The underlying implementation keeps generating paths until one that does not exist is found.
+ * This guarantees that there will be no existing files in the returned directory.
+ *
+ * \param[in] base_name User-specified portion of the created directory
+ * \param[in] parent_path The parent path of the directory that will be created
+ * \return A path to a newly-created directory with base_name and a 6-character unique suffix
+ *
+ * \throws std::system_error If any OS APIs do not succeed.
+ */
+RCPPUTILS_PUBLIC std::filesystem::path create_temporal_directory(
+  const std::string & base_name,
+  const std::filesystem::path & parent_path = temporal_directory_path());
+
+// /**
+//  * \brief Compare two paths for equality.
+//  *
+//  * \return True if both paths are equal as strings.
+//  */
+// RCPPUTILS_PUBLIC bool operator==(const path & a, const path & b);
+// RCPPUTILS_PUBLIC bool operator!=(const path & a, const path & b);
+
+/**
+* \brief Convert the path to a string for ostream usage, such as in logging or string formatting.
+*
+* \param[in] os The stream to send the path string to
+* \param[in] p The path to stringify
+* \return The ostream, for chaining
+*/
+RCPPUTILS_PUBLIC std::ostream & operator<<(std::ostream & os, const std::filesystem::path & p);
```

Comment:
I'm a bit confused about this entire API.  Why can't we just use the existing` rcpputils::fs::create_temp_directory`?

Reply:
There is an ambiguous method call

Reply:
> There is an ambiguous method call

Can you explain more about it?  If we use the fully-qualified version, I'd think that `rcpputils::fs::create_temp_directory` should be sufficient, no?

Reply:
`rcpputils::fs::create_temp_directory` return a rcpputils::fs::path and the idea is to return a `std::filesystem::path`.

 Then the first argument is a string and we got an ambiguous error, that's why I need to change the signature of the function.

from `create_temp_directory` to `create_temporal_directory`

Reply:
> `rcpputils::fs::create_temp_directory` return a rcpputils::fs::path and the idea is to return a `std::filesystem::path`.

Ah, I see!  I missed that fact, thanks.  Now I see why we need another function name.  All right, I have some other comments here, but I'll leave those separately.